packages/openstef-core/src/openstef_core/feature_engineering/validation_transforms/missing_values_transform.py

```diff
@@ -0,0 +1,197 @@
+                to be processed.
+
+        Returns:
+            A pandas DataFrame with trailing null rows removed for the specified
+                features. If no features are configured for checking or none of the
+                configured features exist in the data, returns the original data unchanged.
+        """
+        if not self.config.no_fill_future_values_features:
+            return data
+
+        features_to_check = [f for f in self.config.no_fill_future_values_features if f in data.columns]
+        for feature in self.config.no_fill_future_values_features:
+            if feature not in data.columns:
+                _logger.warning("Feature '%s' not found in dataset columns.", feature)
+
+        if not features_to_check:
+            return data
+
+        subset_df = data[features_to_check]
+        mask = ~subset_df.bfill().isna().any(axis="columns")
+        filtered_data = data.loc[mask]
+
+        return filtered_data
+
+    def fit(self, data: TimeSeriesDataset) -> None:
+        """Fit the missing values transformer to the provided time series data.
+
+        This method prepares the imputer by fitting it to the training data after
+        removing any trailing null rows. The fitted imputer can then be used to
+        transform datasets with missing values.
+
+        Args:
+            data: The time series dataset to fit the transformer on.
+                Trailing null rows will be automatically removed before fitting.
+        """
+        fit_data = self._drop_empty_features(data.data)
+        fit_data = self._remove_trailing_null_rows(fit_data)
+        self.imputer.fit(fit_data)  # pyright: ignore[reportUnknownMemberType]
+
+    def transform(self, data: TimeSeriesDataset) -> TimeSeriesDataset:
+        """Transform the input dataset by removing trailing null rows and imputing missing values.
+
+        This method performs a two-step transformation process:
+        1. Removes trailing rows that contain null values from the dataset
+        2. Applies the configured imputer to fill remaining missing values in the data
+
+        Args:
+            data: The input time series dataset containing data to be transformed.
+                The dataset's data attribute should be a pandas DataFrame or similar structure.
+
+        Returns:
+            The transformed TimeSeriesDataset with trailing null rows removed and
+                missing values imputed. The original dataset structure is preserved with
+                only the data attribute modified.
+        """
+        data_cleaned = self._drop_empty_features(data.data)
+        data_cleaned = self._remove_trailing_null_rows(data_cleaned)
+        data_transformed = self.imputer.transform(data_cleaned)  # pyright: ignore[reportUnknownMemberType, reportUnknownVariableType]
+
+        return TimeSeriesDataset(data=data_transformed, sample_interval=data.sample_interval)  # pyright: ignore[reportArgumentType]
```

Comment:
Most of the typing issues could be solves with proper / explicit typing. It it doesn't work let me know, I might be able to look with you on this.